client/post-editor/post-editor.jsx

```diff
@@ -162,6 +160,10 @@ const PostEditor = React.createClass( {
 		}
 	},
 
+	hideNotice: function() {
+		this.setState( { notice: null } );
```

Comment:
I see below you made a change to treat an empty object as the unset notice. Should we do the same here?